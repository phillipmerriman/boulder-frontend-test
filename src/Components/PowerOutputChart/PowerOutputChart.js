import React from "react";
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
} from "recharts";

import GraphTip from "../GraphTip/GraphTip";

const PowerOutputChart = (props) => {
  const { msAndPower } = props;

  return (
    <>
      <ResponsiveContainer width="100%" aspect={3}>
        <LineChart
          width={500}
          height={300}
          data={msAndPower}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid />
          <XAxis dataKey="millisecondOffset" />
          <YAxis />
          <Tooltip />
          <Line
            type="monotone"
            dataKey="power"
            stroke="#8884d8"
            activeDot={{ r: 8 }}
          />
        </LineChart>
      </ResponsiveContainer>
      <GraphTip />
    </>
  );
};

export default PowerOutputChart;
