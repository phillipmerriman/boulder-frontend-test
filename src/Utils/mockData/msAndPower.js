// mockData to use for testing purposes

const mockData = [
    {
        millisecondOffset: 1000,
        power: 100
    },
    {
        millisecondOffset: 2000,
        power: 200
    },
    {
        millisecondOffset: 3000,
        power: 300
    },
    {
        millisecondOffset: 4000,
        power: 400
    },
    {
        millisecondOffset: 5000,
        power: 500
    },
    {
        millisecondOffset: 6000,
        power: 420
    },
    {
        millisecondOffset: 7000,
        power: 360
    },
    {
        millisecondOffset: 8000,
        power: 400
    },
    {
        millisecondOffset: 9000,
        power: 300
    },
    {
        millisecondOffset: 10000,
        power: 200
    },
];

export default mockData;
