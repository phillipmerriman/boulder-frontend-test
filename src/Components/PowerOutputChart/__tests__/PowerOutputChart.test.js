import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import PowerOutputChart from '../PowerOutputChart';
import mockData from '../../../Utils/mockData/msAndPower';

describe('PowerOutputChart component', () => {
    
    afterEach(cleanup);
    
    it('renders without crashing', () => {
        const div = document.createElement("div");
        ReactDOM.render(<PowerOutputChart />, div);
    });
    
    it('matches snapshot', () => {
        const tree = renderer.create(<PowerOutputChart msAndPower={mockData}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
