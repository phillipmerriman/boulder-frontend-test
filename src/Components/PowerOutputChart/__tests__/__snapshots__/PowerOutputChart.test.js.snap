// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerOutputChart component matches snapshot 1`] = `
Array [
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />,
  <h3
    data-testid="axisInfo"
  >
    X-Axis: Milliseconds | Y-Axis: Power Output
  </h3>,
]
`;
